packages/types/src/payment/provider.ts

```diff
@@ -43,31 +43,31 @@ export type PaymentProviderContext = {
    * The customer's email.
    */
   email?: string
-  /**
-   * The selected currency code.
-   */
-  currency_code: string
-  /**
-   * The payment's amount.
-   */
-  amount: number
   /**
    * The ID of the resource the payment is associated with i.e. the ID of the PaymentSession in Medusa
    */
-  resource_id: string
+  resource_id?: string
   /**
    * The customer associated with this payment.
    */
   customer?: PaymentCustomerDTO
   /**
-   * The context.
+   * The extra fields specific to the provider session.
    */
-  context: { payment_description?: string } & Record<string, unknown>
-  /**
-   * If the payment session hasn't been created or initiated yet, it'll be an empty object.
-   * If the payment session exists, it'll be the value of the payment session's `data` field.
-   */
-  payment_session_data: Record<string, unknown>
+  extra?: Record<string, unknown>
+}
+
+export type CreatePaymentProviderSession = {
+  context: PaymentProviderContext
+  amount: number
+  currency_code: string
+}
+
+export type UpdatePaymentProviderSession = {
+  context: PaymentProviderContext
+  data: Record<string, unknown>
+  amount: number
+  currency_code: string
```

Comment:
comment: Breaking changes. 

I replaced the `context` of the payment provider context DTO to not have `context.context`.

Instead, we now have `extra` which can be used by developers to pass additional non-required details to the payment session creation, e.g. a description of the payment.

Additionally, I extracted the `payment_session_data` of the context to be a standalone property used when creating/updating a payment session.